marlowe/src/Language/Marlowe/Analysis/FSSemantics.hs

```diff
@@ -146,21 +146,34 @@ mkInitialSymState pt (Just State { accounts = accs
 -- The identifiers for Deposit and Choice are calculated using the When clause and
 -- the contract (which is concrete), and using the semantics after a counter example is
 -- found.
-convertToSymbolicTrace :: [(SInteger, SInteger, Maybe SymInput, Integer)] ->
-                          [(SInteger, SInteger, SInteger, SInteger)] -> SBool
-convertToSymbolicTrace [] [] = sTrue
-convertToSymbolicTrace [] ((a, b, c, d):t) = (a .== -1) .&& (b .== -1) .&& (c .== -1) .&&
-                                             (d .== -1) .&& convertToSymbolicTrace [] t
-convertToSymbolicTrace ((lowS, highS, inp, pos):t) ((a, b, c, d):t2) =
+convertRestToSymbolicTrace :: [(SInteger, SInteger, Maybe SymInput, Integer)] ->
+                              [(SInteger, SInteger, SInteger, SInteger)] -> SBool
+convertRestToSymbolicTrace [] [] = sTrue
+convertRestToSymbolicTrace ((lowS, highS, inp, pos):t) ((a, b, c, d):t2) =
   (lowS .== a) .&& (highS .== b) .&& (getSymValFrom inp .== c) .&& (literal pos .== d) .&&
-  convertToSymbolicTrace t t2
+  convertRestToSymbolicTrace t t2
   where
     getSymValFrom :: Maybe SymInput -> SInteger
     getSymValFrom Nothing                       = 0
     getSymValFrom (Just (SymDeposit _ _ _ val)) = val
     getSymValFrom (Just (SymChoice _ val))      = val
     getSymValFrom (Just SymNotify)              = 0
-convertToSymbolicTrace _ _ = error "Provided symbolic trace is not long enough"
+convertRestToSymbolicTrace _ _ = error "Symbolic trace is the wrong length"
+
+isPadding :: [(SInteger, SInteger, SInteger, SInteger)] -> SBool
+isPadding ((a, b, c, d):t) = (a .== -1) .&& (b .== -1) .&& (c .== -1) .&&
+                             (d .== -1) .&& isPadding t
+isPadding [] = sTrue
+
+convertToSymbolicTrace :: [(SInteger, SInteger, Maybe SymInput, Integer)] ->
+                          [(SInteger, SInteger, SInteger, SInteger)] -> SBool
+convertToSymbolicTrace refL symL =
+ let lenRefL = length refL;
```

Comment:
I wonder why semicolon is needed here

Reply:
Good point, it is not, it is needed in Isabelle and I forgot to remove it